### DISCUSSION TITLE
how can I define a ENV variable with different value in github action yaml file?

### DISCUSSION THREAD

**Discussion by bostonlf** [community/community#28427] [Other Feature Feedback, Questions, & Ideas]
example:
I want to define a ENV variable in different value base on different branch name 

      if: ${{ github.ref == 'AA' }}
      env: 
        REACT_APP_BASE_URL: 'aa'
      if: ${{ github.ref == 'BB' }}
      env: 
        REACT_APP_BASE_URL: 'bb'

**Comment by airtower-luna** [comment]
You can't do that within the `env` section for one step. What I'd do is add a separate step that sets the variable, e.g.:

```yaml
- name: Set REACT_APP_BASE_URL
  run: |
    if [ "$GITHUB_REF_NAME" = "AA" ]; then
        echo REACT_APP_BASE_URL=aa >>$GITHUB_ENV
    elif [ "$GITHUB_REF_NAME" = "BB" ]; then
        echo REACT_APP_BASE_URL=bb >>$GITHUB_ENV
    fi
```

Then all further steps in the job can use the environment variable. If you don't want that you could [set a step output](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-setting-a-value) instead of writing to `$GITHUB_ENV`, and then use that in the `env` section of the step(s) that should receive it.

Note I used `GITHUB_REF_NAME` instead of `GITHUB_REF` because it conveniently contains only the branch name, while `GITHUB_REF` comes with the `refs/heads/` prefix for branches. Use whatever fits better.
